src/Dependencies/PooledObjects/PooledHashSet.cs

```diff
@@ -43,4 +43,38 @@ public static PooledHashSet<T> GetInstance()
             return instance;
         }
     }
+
+    internal sealed partial class PooledStack<T> : Stack<T>
```

Comment:
shouldn't need this.  Can just use ArrayBuilder.

Reply:
Not if we are need a pooled collection of `Stack<T>`.